app/services/submission/plant_trial_finalizer.rb

```diff
@@ -26,6 +26,12 @@ def create_new_trait_descriptors
     @new_trait_descriptors = (submission.content.step02.new_trait_descriptors || []).map do |attrs|
       attrs = attrs.with_indifferent_access
       attrs = attrs.merge(common_data)
+      if attrs[:trait].present?
+        attrs[:trait] = Trait.find_by_name(attrs['trait'])
+      end
+      if attrs[:plant_part].present?
```

Comment:
BTW is there a mechanism to do the above "by ids"? I think you were doing it like that somewhere else in the submissions. That is, to have the submission forms display `trait.name` and `plant_part.plant_part`, where needed, but to internally use `trait.id` and `plant_part.id` - so the finders above do not rely on certain column uniqueness.